### ISSUE TITLE
file drop convention is different from desktop geoda

### ISSUE THREAD

**Issue by lanselin** [GeoDaCenter/geoda-ai#1]
The file drop requires all four *.shp, *.shx, *.dbf and *.prj files for a "shape file" input. In desktop GeoDa, only the shp file needs to be dropped. That is a bit of an inconsistency. It would be better if specifying the shp file path also locates the associated files.

**Comment by lixun910** [comment]
This can not be implemented in GeoDa.AI which is a browser app, because the browsers are designed with a sandboxed environment to keep web applications isolated from the local operating system and its file system. 

Accessing files on a user’s device requires explicit permission or action, like drag-n-drop or file upload dialogs, from the user to ensure they are aware of which files are being accessed.

**Comment by lanselin** [comment]
Couldn't there be a way that when only a shp file is dropped, there is a dialog that asks if the remaining files, i.e., shx, dbf and prj can be added?

**Comment by lixun910** [comment]
I think we can not do this as well. Because this will require the browser to access the e.g. dbf and shx files based on the path of dropped shp file. For security reason, the browser cannot access local files on local disk using local file path by default. One needs to select local file by either html upload input element or drag-n-drop. The file access must be used in response to user interactions, such as a button click, to ensure security.

**Comment by lanselin** [comment]
Maybe then enter a warning that all four files must be dropped. This is actually worse than the desktop
version from a user’s perspective and is actually a royal pain in the neck. It would be nice if this could be
avoided. Let’s find a creative solution. Just saying it can’t be done doesn’t do it for me.

L.

**Comment by lixun910** [comment]
Yes, I agree.  There is a warning the all files must be dropped, see the screenshot below. Maybe we need a better design to tell the user why and how.

<img width="779" alt="Screenshot 2024-10-28 at 2 49 11 PM" src="https://github.com/user-attachments/assets/4bc11a10-1ef9-4453-a73d-1becd320cebf">

**Comment by lixun910** [comment]
Btw, if we turn the https://geoda.ai into a native desktop app, there will be no such a restriction.

**Comment by lixun910** [comment]
Fixes in https://github.com/GeoDaCenter/reactgeoda/pull/189
Fixes are available at https://geoda.ai/preview
